packages/menu/src/ui-menu.tsx

```diff
@@ -55,10 +74,44 @@ export const UiMenu: React.FC<UiMenuProps> = ({ children, closeMenuCB, visible }
     };
   }, [escCB]);
 
+  React.useEffect(() => {
+    if (visible && isSmall && !isOpen) {
+      actions.openDialog();
+    }
+
+    if (visible && isOpen && !isSmall) {
+      actions.closeDialog();
+    }
+  }, [visible, isOpen, isSmall]);
+
   if (!visible) {
     return null;
   }
 
+  if (fullscreenOnSmall) {
+    return (
+      <>
+        <UiViewport criteria={Breakpoints.SMALL}>
+          <UiDialog dialogId={dialogId} handleDialogClose={closeMenuCB} type={UiDialogType.FULLSCREEN}>
```

Comment:
This dialog is shown when the breakpoint is small and is what makes the fullscreen menu to be rendered.